### PR TITLE
Various

### PR DESCRIPTION
**Pull request by just1689**
- Persist
- SSL
- Probably other things.
- Much of this needs refactoring

### CODE REVIEW

**Comment by just1689** [comment]
Just a heads up - you can ignore the travis failing - it relates to a known issue in Go https://github.com/golang/go/issues/29278 Lots of people are having to delete the go.sum file on their CI server before a build

File: README.md [hunk 1]
```diff
@@ -4,58 +4,102 @@
 [![](https://images.microbadger.com/badges/version/planckscloud/plancks-cloud.svg)](https://microbadger.com/images/planckscloud/plancks-cloud "Get your own version badge on microbadger.com")&nbsp;<a href="https://trello.com/b/NutXeZwS/plancks-roadmap"><img src="https://img.shields.io/badge/Roadmap-Trello-brightgreen.svg" /></a>
 <a href="https://coggle.it/diagram/XEgmhoO3UopF8htc/t/logo"><img src="https://img.shields.io/badge/Ideas-Coggle-brightgreen.svg" /></a>&nbsp;[![License](http://img.shields.io/:license-mit-blue.svg?style=flat)](http://badges.mit-license.org)
 
-# What is Plancks Cloud?
+# What is Planck's Cloud?
 
-Planck's Cloud is a private or public cloud that aims to turn every home into a datacenter. Host your next project from your own home with Planck's Cloud.
+Have you ever wanted to build a tiny app and host it from home? Have you ever had a startup idea and wanted to deploy it within an hour? Planck's Cloud aims to make deploying, reverse proxying and HTTPS your next idea from home a walk in the park.
```
Comment (connavar) [inline README.md]: Planck's Cloud aims to make deploying, reverse proxying and HTTPS-ing your next idea from home a walk in the park.

File: README.md [hunk 2]
```diff
@@ -4,58 +4,102 @@
...
 
 # How it works
 
-Planck's Cloud allows you to run containers on your network at home and make endpoints available on the public Internet.
+Install Planck's Cloud onto a machine you want to use as a server. Install the CLI and start deploying!
```
Comment (connavar) [inline README.md]:
```suggestion
Install Planck's Cloud onto a machine you want to use as a server. Simply install the CLI and start deploying! 
```

File: README.md [hunk 3]
```diff
@@ -4,58 +4,102 @@
...
 
-# Current Status for Plancks Cloud?
+# Current Status of Planck's Cloud?
+<img src="https://goreportcard.com/badge/github.com/plancks-cloud/plancks-cloud">&nbsp;<a href="https://codeclimate.com/github/plancks-cloud/plancks-cloud/maintainability"><img src="https://api.codeclimate.com/v1/badges/81aff827de3938808c2d/maintainability" /></a>&nbsp;[![codebeat badge](https://codebeat.co/badges/25407218-e856-4f5e-ac7c-9d045dc0fe5a)](https://codebeat.co/projects/github-com-plancks-cloud-plancks-cloud-master)
 
-Currently the plan for Planck's Cloud is to make it possible for anyone and everyone to be able to easily spin up their own Planck's Cloud network from any hardware that is available, and use it to run their projects.
+The features currently available are:
+- Create, update and delete docker services.
+- Create routes for ingress. Route traffic for a hostname to a service or machine on your network.
+- SSL offloading. Expose endpoints with LetEncrypt provided HTTPS.
```
Comment (connavar) [inline README.md]:
```suggestion
- SSL offloading. Expose endpoints with LetsEncrypt provided HTTPS.
```

File: README.md [hunk 4]
```diff
@@ -4,58 +4,102 @@
...
 
+# Architecture
 
-# Architecture and Design
-<img src="https://goreportcard.com/badge/github.com/plancks-cloud/plancks-cloud">&nbsp;<a href="https://codeclimate.com/github/plancks-cloud/plancks-cloud/maintainability"><img src="https://api.codeclimate.com/v1/badges/81aff827de3938808c2d/maintainability" /></a>&nbsp;[![codebeat badge](https://codebeat.co/badges/25407218-e856-4f5e-ac7c-9d045dc0fe5a)](https://codebeat.co/projects/github-com-plancks-cloud-plancks-cloud-master)
+Planck's Cloud runs is an Open Source Golang app that runs inside a docker container. The standard method of deployment is to run the container as a service inside a docker swarm. Planck's Cloud communicates with the docker daemon to create, update and delete services. Docker also proxies traffic from 80 and 443 on the host to the Planck's Cloud container. Planck's Cloud matches DNS entries to reverse proxy traffic to various services hosted in the swarm or other computers on your network.
```
Comment (connavar) [inline README.md]:
```suggestion
Planck's Cloud is an open source Golang app that runs inside a docker container. The standard method of deployment is to run the container as a service inside a docker swarm. Planck's Cloud communicates with the docker daemon to create, update and delete services. Docker also proxies traffic from 80 and 443 on the host to the Planck's Cloud container. Planck's Cloud matches DNS entries to reverse proxy traffic to various services hosted in the swarm or other computers on your network.
```

File: controller/route.go [hunk 5]
```diff
@@ -10,12 +11,12 @@ import (
 )
 
 var (
-	proxy = flag.String("proxy", ":6228", "TCP address to listen to")
+	proxy = flag.String("proxy", ":80", "TCP address to listen to")
```
Comment (connavar) [inline controller/route.go]:
```suggestion
	proxy = flag.String("proxy", ":80", "TCP port to listen on")
```

**Pull request merged**